rolereact/rolereact.go

```diff
@@ -0,0 +1,840 @@
+			Components: []component.Component{
+				&component.TextDisplay{Content: "Impossible de trouver la modification de message. Veuillez réessayer."},
+			},
+		}
+	} else {
+		role.RoleID = data.Values[0]
+		responseData = MessageModifyRoleData(i, parameters, "")
+	}
+	err := s.InteractionAPI().Respond(i.Interaction, &interaction.Response{
+		Type: types.InteractionResponseUpdateMessage,
+		Data: &responseData,
+	})
+	if err != nil {
+		s.Logger().Error("Unable to send open reaction role message", "error", err)
+	}
+}
+
+func HandleSetReaction(
+	s bot.Session,
+	i *event.InteractionCreate,
+	data *interaction.MessageComponentData,
+	parameters *EditIDWithRole,
+	resp *cmd.ResponseBuilder,
+) {
+	message, ok := GetMessageFromEditID(i, parameters.MessageEditID)
+	var role *config.RoleReact
+	if ok {
+		roleIndex := slices.IndexFunc(message.Roles, func(role *config.RoleReact) bool { return role.CounterID == parameters.RoleCounterID })
+		if roleIndex != -1 {
+			role = message.Roles[roleIndex]
+		}
+	}
+	if !ok || role == nil {
+		err := s.InteractionAPI().Respond(i.Interaction, &interaction.Response{
+			Type: types.InteractionResponseUpdateMessage,
+			Data: &interaction.ResponseData{
+				Flags: channel.MessageFlagsEphemeral | channel.MessageFlagsIsComponentsV2,
+				Components: []component.Component{
+					&component.TextDisplay{Content: "Impossible de trouver la modification de message. Veuillez réessayer."},
+				},
+			},
+		})
+		if err != nil {
+			s.Logger().Error("Unable to send open reaction role message", "error", err)
+		}
+		return
+	}
+	responseData := MessageModifyRoleData(i, parameters, "Ajoute la réaction que tu veux choisir au message de rôle de réaction (tu peux y accéder avec le bouton ci-dessous)")
+	s.InteractionAPI().Respond(i.Interaction, &interaction.Response{
+		Type: types.InteractionResponseUpdateMessage,
+		Data: &responseData,
+	})
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+
+	emojiChann := make(chan emoji.Emoji)
+
+	cancelHandler := s.EventManager().AddHandler(func(s bot.Session, e *event.MessageReactionAdd) {
+		if e.MessageID == message.MessageID && e.UserID == i.Member.User.ID {
+			emojiChann <- e.Emoji
+		}
+	})
+	defer cancelHandler()
+
+	select {
+	case emoji := <-emojiChann:
+		emojiName := emoji.APIName()
+		err := s.ChannelAPI().MessageReactionAdd(message.ChannelID, message.MessageID, emojiName)
+		if err != nil {
+			responseData := MessageModifyRoleData(i, parameters, "La réaction n'est pas utilisable. Cela peut être résolu en l'ajoutant à ce serveur")
+			err = s.InteractionAPI().Respond(i.Interaction, &interaction.Response{
+				Type: types.InteractionResponseUpdateMessage,
+				Data: &responseData,
+			})
+			if err != nil {
+				s.Logger().Error("Unable to edit original response", "error", err)
+			}
+			return
+		}
+		s.ChannelAPI().MessageReactionRemove(message.ChannelID, message.MessageID, emojiName, i.Member.User.ID)
```

Comment:
Error not checked here.

Reply:
This is not really important, so it doesn't matters if it fails.